src/scrapeFieldRainoutInfo.ts

```diff
@@ -0,0 +1,167 @@
+   * with newer entries appearing earlier in the list if it is not reversed. Reversing it
+   * allows the newer entries to override the older ones.
+   *
+   * This schedule goes back to 2015, so we slice it because we really only care about
+   * the most recent entries.
+   *
+   * This should cover most of the common cases. It's not really possible to do much with the "check back later" ones.
+   *
+   * If the value for the date is true, the field is rained out.
+   */
+  return rainoutInfo
+    .slice(0, limit)
+    .reverse()
+    .reduce(
+      (acc, row) => {
+        const splitDate = row[0].split("/");
+        const date = `${splitDate[2]}-${splitDate[0]}-${splitDate[1]}`;
+
+        const fieldComplex = row[2].toLowerCase();
+        const fieldStatus = row[3].toLowerCase();
+        const additionalInformation = row[4];
+        const additionalInformationArr =
+          additionalInformation && additionalInformation.trim() !== ""
+            ? additionalInformation
+                .split("\n")
+                .map((entry) => entry.trim().toLowerCase())
+            : [];
+
+        acc[date] = false; // assume the field is not rained out
+
+        if (
+          fieldComplex === FieldComplex.ALL_GRASS_FIELDS_AND_DIAMONDS &&
+          CLOSED_STATUSES.includes(fieldStatus as ClosedFieldStatus)
+        ) {
+          // the easy case: all of the fields are closed
+          acc[date] = true;
+        } else if (
+          fieldComplex === FieldComplex.OTHER_FIELDS ||
+          fieldComplex === FieldComplex.OTHER_SEE_BELOW
+        ) {
+          // second most common case: some subset of fields is explicitly listed as open or closed
+          if (additionalInformationArr.includes("polo")) {
+            const maybeFieldStatus = additionalInformationArr[0].replace(
+              /\W/g,
+              "",
+            );
+            if (
+              CLOSED_STATUSES.includes(fieldStatus as ClosedFieldStatus) ||
+              CLOSED_STATUSES.includes(maybeFieldStatus as ClosedFieldStatus)
+            ) {
+              acc[date] = true;
+            } else if (
+              OPEN_STATUSES.includes(fieldStatus as OpenFieldStatus) ||
+              OPEN_STATUSES.includes(maybeFieldStatus as OpenFieldStatus)
+            ) {
+              acc[date] = false;
+            }
+          }
+        } else if (fieldComplex === FieldComplex.POLO_FIELDS) {
+          // the polo fields are specifically listed as open/closed - pretty uncommon
+          if (CLOSED_STATUSES.includes(fieldStatus as ClosedFieldStatus)) {
+            acc[date] = true;
+          } else if (OPEN_STATUSES.includes(fieldStatus as OpenFieldStatus)) {
+            acc[date] = false;
+          }
+        } else if (fieldComplex === FieldComplex.ALL_FIELDS_EXCEPT) {
+          // this status reverses whatever the field status is for the listed field
+          if (additionalInformationArr.includes("polo")) {
```

Comment:
is it ever not `FieldComplex.POLO_FIELDS`?

Reply:
It's a variety of things, but we only care about a subset